osc/osc.go

```diff
@@ -720,70 +716,70 @@ func readArguments(msg *Message, reader *bufio.Reader, start *int) error {
 
 		case 'i': // int32
 			var i int32
-			if err = binary.Read(reader, binary.BigEndian, &i); err != nil {
+			if err = binary.Read(buffer, binary.BigEndian, &i); err != nil {
 				return err
 			}
 			*start += 4
-			msg.Append(i)
+			msg.Arguments = append(msg.Arguments, i)
```

Comment:
Doing this seems to be cheaper, but I'm not 100% sure if it is, needs more testing.